src/main/java/io/pinecone/clients/Index.java

```diff
@@ -2,23 +2,32 @@
 
 import com.google.protobuf.Struct;
 import io.pinecone.commons.PineconeDataPlaneInterface;
+import io.pinecone.configs.PineconeConnection;
 import io.pinecone.exceptions.PineconeValidationException;
 import io.pinecone.proto.*;
 import io.pinecone.unsigned_indices_model.QueryResponseWithUnsignedIndices;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.List;
+import java.util.concurrent.TimeUnit;
 
-public class PineconeBlockingDataPlaneClient implements PineconeDataPlaneInterface<UpsertResponse,
-        QueryResponseWithUnsignedIndices, FetchResponse, UpdateResponse, DeleteResponse, DescribeIndexStatsResponse> {
+public class Index implements PineconeDataPlaneInterface<UpsertResponse,
```

Comment:
To prevent memory leakage issues, @ssmith-pc made a great suggestion to add Pinecone connection as a member of this class and implement AutoClosebale interface. We will document the handling of connections better.